Week3/homework/src/actions/createNewTodo.js

```diff
@@ -0,0 +1,35 @@
+'use-strict';
+
+const connection = require('../databaseConfig');
+const nanoid = require('nanoid');
+
+const createNewTodo = (req, res) => {
+  if (req.body.todo === '' || req.body.todo === null) {
+    res.status(400);
+    res.json('Please define a todo!');
+  } else {
+    connection.query(
+      `INSERT INTO todos VALUES(
+        '${req.params.todoListId}',
+        '${nanoid()}',
+        '${req.body.todo}',
+        '${req.body.done}',
+        STR_TO_DATE( '${req.body.dueDate}', '%d/%m/%Y %H:%i:%s'),
+        '${req.body.tag}'
+        )`
+    );
+    connection.query(
+      `SELECT * FROM todos WHERE ToDoList_Id = '${req.params.todoListId}'`,
```

Comment:
Look, here is the problem.

JS has the event loop: https://flaviocopes.com/javascript-event-loop/
What can happen here: 
- if INSERT query takes too long, let's say, the DB is heavy loaded
- then JS will run SELECT query immediately, without waiting for INSERT query. 
- it will raise an error

One of the solutions - to use callbacks or promises. The goal is - to respect the event loop anyway

Reply:
Yes, I didn't take into consideration the event loop while I wrote the code because the database was not heavy. But I should have. I made the corrections for it. Thanks :pray: